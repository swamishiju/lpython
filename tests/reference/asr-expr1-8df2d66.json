{
    "basename": "asr-expr1-8df2d66",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/expr1.py",
    "infile_hash": "a0c9e93e40f29b7d4541f8c513de490a892abf78a34700478d1827dd",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-expr1-8df2d66.stdout",
    "stdout_hash": "1e465939c64050a35b58af65eb027f1e7e9645fb02a02d3f9c9afa0c",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}