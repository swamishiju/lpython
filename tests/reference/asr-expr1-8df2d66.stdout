(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            test_namedexpr:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            a:
                                                (Variable
                                                    3
                                                    a
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    3
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            y:
                                                (Variable
                                                    3
                                                    y
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    test_namedexpr
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(Assignment
                                        (Var 3 x)
                                        (NamedExpr
                                            (Var 3 y)
                                            (IntegerConstant 0 (Integer 4) Decimal)
                                            (Integer 4)
                                        )
                                        ()
                                        .false.
                                        .false.
                                    )
                                    (If
                                        ()
                                        (NamedExpr
                                            (Var 3 a)
                                            (StringOrd
                                                (StringConstant
                                                    "3"
                                                    (String 1 (IntegerConstant 1 (Integer 4) Decimal) ExpressionLength DescriptorString)
                                                )
                                                (Integer 4)
                                                (IntegerConstant 51 (Integer 4) Decimal)
                                            )
                                            (Integer 4)
                                        )
                                        [(Assignment
                                            (Var 3 x)
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            ()
                                            .false.
                                            .false.
                                        )]
                                        []
                                    )
                                    (WhileLoop
                                        ()
                                        (NamedExpr
                                            (Var 3 a)
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            (Integer 4)
                                        )
                                        [(Assignment
                                            (Var 3 y)
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            ()
                                            .false.
                                            .false.
                                        )]
                                        []
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    ()
                    []
                    .false.
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        4
                        {
                            
                        })
                    main_program
                    []
                    []
                )
        })
    []
)
